### ISSUE TITLE
Check ElasticSearch version

### ISSUE THREAD

**Issue by vegaed** [thinkaurelius/titan#338]
Titan should check the version number for ElasticSearch on startup and if not compatible fail.

**Comment by mbroecheler** [comment]
That's a good point. What (types of) errors to do you see when the versions
are incompatible?
Thanks,
Matthias

On Tue, Jul 30, 2013 at 5:14 AM, Edgardo Vega notifications@github.comwrote:

> Titan should check the version number for ElasticSearch on startup and if
> not compatible fail.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/thinkaurelius/titan/issues/338
> .

## 

Matthias Broecheler
http://www.matthiasb.com

**Comment by vegaed** [comment]
Matthias, 

I just ran into this issue, hence the suggestion. In summary when I was trying to use data with geopoints gremlin would fail with Could not commit transaction due to exception during persistence
Display stack trace? [yN] y

Here is the link  forum with all the details. https://groups.google.com/forum/#!topic/aureliusgraphs/V9poQr6WOkM

Cheers,

Edgardo

**Comment by dalaro** [comment]
I would like to check both the client and server version, but I couldn't find a way to retrieve the server ES version using the Java API.  This is terribly easy to do with the HTTP REST API -- just GET the root -- so I'm kind of wondering if it's in the Java API and I just missed it.

The check does _not_ cause failure.  It just logs a warning.  We could make it throw a RuntimeException, but I didn't want to be that restrictive.

Bottom line: I'm only checking the client ES version.  The runtime version is compared with the compile-time version in a manner very similar to TitanConstants: write a maven pom.xml property to a resource file, load it, then compare the loaded value against a runtime value.
